bpo-45163: haiku build fix proposal.

devnexen opened the pull request:
linkage issues mainly for shared libs and missing system library,
 also little nit into the signal extension as strsignal returns
a constant in this platform.

<!-- issue-number: [bpo-45163](https://bugs.python.org/issue45163) -->
https://bugs.python.org/issue45163
<!-- /issue-number -->

serhiy-storchaka commented on Modules/signalmodule.c:
```diff
@@ -663,7 +667,7 @@ signal_strsignal_impl(PyObject *module, int signalnum)
 #else
     errno = 0;
-    res = strsignal(signalnum);
+    res = (const char *)strsignal(signalnum);
```
It is not needed. `char*` is automatically converted to `const char*`.
```suggestion
    res = strsignal(signalnum);
```

serhiy-storchaka commented on Modules/signalmodule.c:
```diff
@@ -611,7 +611,11 @@ static PyObject *
+    const char * res;
+#else
     char *res;
+#endif
```
Just makes it always `const`. It is more correct semantically, because `res` is initialized with pointers to string literals and the content is not changed.
```suggestion
    const char *res;
```

serhiy-storchaka committed:
Apply suggestions from code review

devnexen committed:
haiku build fix proposal.

linkage issues mainly for shared libs and missing system library,
 also little nit into the signal extension as strsignal returns
a constant in this platform.

blurb-it[bot] committed:
📜🤖 Added by blurb_it.
